### PR TITLE
Enable expression and protein data on multi-study when configurable condition is met

### PR DESCRIPTION
**Pull request by alisman**

### CODE REVIEW

File: src/shared/components/query/MolecularProfileSelector.tsx [hunk 1]
```diff
@@ -121,6 +134,17 @@ export default class MolecularProfileSelector extends QueryStoreComponent<
         groupLabel: string
     ) {
         let profiles = this.store.getFilteredProfiles(molecularAlterationType);
+
+        const groupedProfiles = _.groupBy(profiles, profile => {
+            const profileType = profile.molecularProfileId.replace(
+                new RegExp(profile.studyId + '_'),
+                ''
+            );
```
Comment (onursumer) [inline src/shared/components/query/MolecularProfileSelector.tsx:142]: should we add a comment here to explain why we are removing the study id?

File: src/shared/components/query/QueryStore.ts [hunk 2]
```diff
@@ -1556,14 +1565,16 @@ export class QueryStore {
     getFilteredProfiles(
         molecularAlterationType: MolecularProfile['molecularAlterationType']
     ) {
-        return this.molecularProfilesInSelectedStudies.result.filter(
+        const ret = this.molecularProfilesInSelectedStudies.result.filter(
```
Comment (onursumer) [inline src/shared/components/query/QueryStore.ts:1568]: Do we really need to declare a new const `ret`? We just return `ret` without doing anything else, right?

**Review by onursumer — COMMENTED** [review]
Looks good to me. Just added couple minor comments.

**Comment by SRodenburg** [comment]
should address related issue:
https://github.com/cBioPortal/cbioportal/issues/9865

**Pull request merged**